### DISCUSSION TITLE
Domain BaseEvent is dependent on MediatR INotification

### DISCUSSION THREAD

**Discussion by NghiaNguyen170192** [jasontaylordev/CleanArchitecture#707] [Q&A]
Hi, 
I followed the CleanArchitecture for a while and noticed that BaseEvent was independent from MediatR INotification on https://github.com/jasontaylordev/CleanArchitecture/issues/199 and https://github.com/jasontaylordev/CleanArchitecture/pull/217/

And I tried to learn the CleanArchitecture design and saw this commit https://github.com/jasontaylordev/CleanArchitecture/commit/03862a3617bd072aa56980591650962c5aa15c26 reverted to make BaseEvent become dependent to INotification again. 

May I ask why making the Domain layer depends on external library like MediatR?

**Comment by jasontaylordev** [comment]
MediatR is lightweight, free from other dependencies, and very useful. It's used in the Domain to support Domain events. I'm quite happy with this usage and implementation. 

I think Jimmy covers these concerns quite well in [You Probably Don't Need to Worry About MediatR](https://jimmybogard.com/you-probably-dont-need-to-worry-about-mediatr/).
